metricbeat/module/kubernetes/test_kubernetes.py

```diff
@@ -4,55 +4,362 @@
+        volumeMounts:
+        - name: mnt
+          mountPath: /mnt
+  volumeClaimTemplates:
+  - metadata:
+      name: mnt
+    spec:
+      accessModes: ["ReadWriteOnce"]
+      resources:
+        requests:
+          storage: 1Mi 
+"""
 
-    # Tests are disabled as current docker-compose settings fail to start in many cases:
-    # COMPOSE_SERVICES = ['kubernetes']  # 'kubestate']
+DEPLOYMENT_MANIFEST = """\
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: basic-deployment
+  labels:
+    app: basic-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: basic-deployment
+  template:
+    metadata:
+      labels:
+        app: basic-deployment
+    spec:
+      containers:
+      - name: sh
+        image: alpine:3
+        command: ["sh", "-c", "sleep infinity"]
+"""
 
-    @unittest.skipUnless(False and metricbeat.INTEGRATION_TESTS, "integration test")
+CRONJOB_MANIFEST = """\
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: basic-cronjob
+spec:
+  schedule: "* * * * *"
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          name: basic-job
+        spec:
+          containers:
+          - name: hello
+            image: alpine:3
+            command: ["sh", "-c", "echo Hello!"]
+          restartPolicy: Never
+"""
+
+RESOURCE_QUOTA_MANIFEST = """\
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: object-counts
+spec:
+  hard:
+    configmaps: "99"
+"""
+
+class Test(metricbeat.BaseTest, KindMixin):
+
+    # Map kube-state-metrics port from inside of kind to host.
+    KIND_EXTRA_PORT_MAPPINGS = [
+        {
+            "containerPort": 30808,
+            "hostPort": 30808,
+        }
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.kind_create_cluster(cls.build_path)
+        if metricbeat.INTEGRATION_TESTS and cls.kind_available():
+            try:
+                cls.kind_kubectl(cls.build_path, ["apply", "-f", "-"], input=KUBE_STATE_METRICS_MANIFEST)
+                cls.kind_kubectl(cls.build_path, ["apply", "-f", "-"], input=STS_MANIFEST)
+                cls.kind_kubectl(cls.build_path, ["wait", "--timeout=300s",
+                                                  "--for=condition=available", "deployment/kube-state-metrics"])
+                cls.kind_kubectl(cls.build_path, ["wait", "--timeout=300s",
+                                                  "--for=condition=ready", "pod/basic-sts-0"])
+            except:
+                cls.kind_delete_cluster(cls.build_path)
+                raise
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        cls.kind_delete_cluster(cls.build_path)
+
+    @classmethod
+    def kube_state_metrics_hosts(cls):
+        """
+        kube-state-metrics host connection.
+        """
+        return ["localhost:30808"]
+
+    @unittest.skipUnless(KindMixin.kind_available() and metricbeat.INTEGRATION_TESTS, "integration test")
```

Comment:
We can also use `TEST_TAGS` (see https://github.com/elastic/beats/pull/17075) to select when to run these tests. This way we avoid skipping the tests by mistake because `kind` is not properly installed in some CI worker.
```suggestion
    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
    @metricbeat.tag('kubernetes')
```
These tests will need to be run then with `TEST_TAGS=kubernetes`